src/parser/ad_parser.js

```diff
@@ -237,47 +236,70 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
   extensions.forEach(extNode => {
-    const ext = new AdExtension();
-    const extNodeAttrs = extNode.attributes;
-    const childNodes = extNode.childNodes;
+    const ext = parseExtension(extNode);
 
-    if (extNode.attributes) {
-      for (const extNodeAttrKey in extNodeAttrs) {
+    if (ext) {
+      collection.push(ext);
+    }
+  });
+}
+
+/**
+ * Parses an extension child node
+ * @param {Object} extNode - The extension node to parse
+ * @return {Object}
+ */
+function parseExtension(extNode) {
+  // Ignore comments
+  if (extNode.nodeName === '#comment') return null;
+
+  const ext = new AdExtension();
+  const extNodeAttrs = extNode.attributes;
+  const childNodes = extNode.childNodes;
+  const txt = parserUtils.parseNodeText(extNode);
+
+  if (txt !== '') {
```

Comment:
I think we should only add value, if `extNode` doesn't have any children, otherwise it will give unreadable results